crates/llm-ls/src/main.rs

```diff
@@ -398,37 +402,34 @@ fn build_prompt(
 
 async fn request_completion(
     http_client: &reqwest::Client,
-    ide: Ide,
-    model: &str,
-    request_params: RequestParams,
-    api_token: Option<&String>,
     prompt: String,
+    params: CompletionParams,
 ) -> Result<Vec<Generation>> {
     let t = Instant::now();
+    let model = params.model.clone();
+    let adaptor = params.adaptor.clone();
+    let api_token = params.api_token.clone();
+    let ide = params.ide.clone();
+
+    let json = Adaptors.adapt_body(prompt, params);
+    let headers = Adaptors.adapt_headers(adaptor.clone(), api_token.as_ref(), ide)?;
     let res = http_client
-        .post(build_url(model))
-        .json(&APIRequest {
-            inputs: prompt,
-            parameters: request_params.into(),
-        })
-        .headers(build_headers(api_token, ide)?)
+        .post(build_url(&model))
+        .json(&json)
+        .headers(headers)
         .send()
         .await
         .map_err(internal_error)?;
 
-    let generations = match res.json().await.map_err(internal_error)? {
-        APIResponse::Generation(gen) => vec![gen],
-        APIResponse::Generations(gens) => gens,
-        APIResponse::Error(err) => return Err(internal_error(err)),
-    };
+    let generations = Adaptors.adapt_blob(adaptor, res.text().await);
```

Comment:
```suggestion
    let generations = Adaptors.adapt_blob(adaptor, res.text().await)?;
```

Reply:
Is this needed? Adding it causes `generations` to unwrap from the result, which I need to rewrap in `Ok` anyway, since `?` requires the return type to be `Result`